src/main/pending_query_result.cpp

```diff
@@ -74,4 +75,14 @@ void PendingQueryResult::Close() {
 	context.reset();
 }
 
+bool PendingQueryResult::IsFinished(PendingExecutionResult result) {
+	if (result == PendingExecutionResult::RESULT_NOT_READY) {
```

Comment:
Nit: I think it's nicer to formulate this method positively in terms of what finished means, i.e.

```
if (result == PendingExecutionResult::RESULT_READY || 
    result == PendingExecutionResult::EXECUTION_ERROR) {
  return true;
} 
return false;
```

Reply:
sounds good!